src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -3302,19 +3306,37 @@ void UpdateZoneSizing(EnergyPlusData &state, Constant::CallIndicator const CallI
                 }
             }
 
-            writeZszSpsz(state,
-                         state.files.zsz,
-                         state.dataGlobal->NumOfZones,
-                         state.dataZoneEquip->ZoneEquipConfig,
-                         state.dataSize->CalcFinalZoneSizing,
-                         state.dataSize->CalcZoneSizing);
+            // Write zone sizing (zsz) and space sizing (spsz) outputs
+            if (state.dataSize->SizingFileColSep == DataStringGlobals::CharComma) {
+                state.files.zsz.filePath = state.files.outputZszCsvFilePath;
+            } else if (state.dataSize->SizingFileColSep == DataStringGlobals::CharTab) {
+                state.files.zsz.filePath = state.files.outputZszTabFilePath;
+            } else {
+                state.files.zsz.filePath = state.files.outputZszTxtFilePath;
+            }
+            state.files.zsz.ensure_open(state, "UpdateZoneSizing", state.files.outputControl.zsz);
+
+            bool forSpaces = false;
+            writeZszSpsz(
+                state, state.files.zsz, state.dataGlobal->NumOfZones, state.dataSize->CalcFinalZoneSizing, state.dataSize->CalcZoneSizing, forSpaces);
+
             if (state.dataHeatBal->doSpaceHeatBalanceSizing) {
+                if (state.dataSize->SizingFileColSep == DataStringGlobals::CharComma) {
+                    state.files.spsz.filePath = state.files.outputSpszCsvFilePath;
+                } else if (state.dataSize->SizingFileColSep == DataStringGlobals::CharTab) {
+                    state.files.spsz.filePath = state.files.outputSpszTabFilePath;
+                } else {
+                    state.files.spsz.filePath = state.files.outputSpszTxtFilePath;
+                }
+                state.files.spsz.ensure_open(state, "UpdateZoneSizing", state.files.outputControl.spsz);
```

Comment:
Wait until here to open zsz and spsz files. At this point we know if `doSpaceHeatBalanceSizing` is active.

Reply:
Actually glad you fixed this. I saw the 0 kB files but wasn't sure how to avoid that. This is a perfect fix.